reconstruct/reconstruct.go

```diff
@@ -101,9 +105,95 @@ func ReconAccuracy(simFilename string, reconFilename string, leavesOnlyFile stri
 	acc := 100 - accuracy
 	answer[allNodes] = acc
 
+	return answer, accByBase
+}
+
+// ReconAccuracyByBase will run if the option given to ReconAccuracy = 1. This function calculates the percentage of first, second and third bases in codons that were correct.
+func ReconAccuracyByBase(simFilename string, reconFilename string, gpFilename string) map[string][]float64 {
+	sim := fasta.Read(simFilename)
+	recon := fasta.Read(reconFilename)
+	genes := genePred.Read(gpFilename)
+	answer := make(map[string][]float64)
+
+	for s := 0; s < len(sim); s++ {
+		for r := 0; r < len(recon); r++ {
```

Comment:
I believe there is a "fasta map" in the fasta package which will let you lookup a sequence by its name.  This will also let you realize when that sequence is not found.  A possible update would be to read "recon" into a map so that you can just lookup sim[s].Name in that map.  This will let you removed the "r" loop and also give an error if it is not found.